replica/replica.go

```diff
@@ -152,5 +152,33 @@ func (replica *replica) shouldProposeBlock() bool {
 
 func (replica *replica) generateBlock() block.Block {
```

Comment:
This implementation is wrong, but the `sharding` branch (and its upcoming PR) already address this issue so we can leave it like this for now.